### PR TITLE
Project Initialization with Vite (Local & Remote)

### LINKED ISSUE #4: Project Initialization with Vite (Local & Remote)
### Description
As a Developer, I should be able to set up a new React project with TypeScript using Vite for a fast development environment, both locally and in a remote repository, so that I can efficiently begin development with the necessary tools and libraries in place.

#### Project Initialization
1. Set up a new React project with TypeScript using Vite.
- [x]  Create a new Vite project with the appropriate template for React and TypeScript.
- [x]  Configure the project to work locally on the developer's machine.

#### Install Essential Dependencies
2. Install and configure essential libraries:
   - `react-router-dom`: For handling routing in the application.
   - `axios`: For making HTTP requests.
   - `redux`, `react-redux`, `@reduxjs/toolkit`: For state management.
   - `eslint`: For linting JavaScript and TypeScript code to ensure code quality.
   - `prettier`: For code formatting.
   - `vite`: For building and serving the project.
   - `vitest`: For unit testing.
   - `@testing-library/react`, `@testing-library/jest-dom`: For testing React components.

### Developer Notes

- Ensure the project is configured to use TypeScript.
- Configure ESLint and Prettier to enforce coding standards and format the code consistently.
- Set up scripts in `package.json` for running the development server, building the project, and running tests.
- Add a `.gitignore` file to exclude unnecessary files from the repository.
-  Ensure that all dependencies are installed and configured correctly.
- Ensure that the project can be built and run both locally and in the remote repository.

### PR DESCRIPTION
**Pull request by wayneleon1**
### What does this PR do?

This PR sets up a new React project with TypeScript using Vite for a fast development environment, both locally and in a remote repository. It also installs essential libraries such as `react-router-dom`, `axios`, `redux`, `react-redux`, `@reduxjs/toolkit`, `eslint`, `prettier`, `vite`, `vitest`, `@testing-library/react`, and `@testing-library/jest-dom`.

### Description of Task to be Completed?
- [x] Initialize a new React project with TypeScript using Vite for enhanced performance and developer experience.
- [x] Install and configure essential libraries and tools for the project, including routing (`react-router-dom`), HTTP requests (`axios`), state management (`redux`, `react-redux`, `@reduxjs/toolkit`), linting (`eslint`), formatting (`prettier`), build tool (`vite`), testing (`vitest`, `@testing-library/react`, `@testing-library/jest-dom`).

### How should this be manually tested?
1. **Clone the Repository:**
   - Clone the project repository to your local machine.
2. **Install Dependencies:**
   - Run `npm install` to install all listed dependencies.
3. **Start the Development Server:**
   - Execute `npm run dev` to start the Vite development server.
4. **Test Application Locally:**
   - Open the application in a web browser to verify its functionality.
5. **Run Tests:**
   - Use `npm run test` to execute tests written with Vitest and Testing Library utilities.

### Any background context you want to provide?
- The project aims to leverage Vite for its speed and efficiency in setting up a React project with TypeScript, enhancing the development workflow.
- Installation of essential dependencies ensures that the project has robust features like routing, state management, and testing support right from the start.
- This setup facilitates both local and remote development, allowing for seamless collaboration among team members

### CODE REVIEW

**Review by EddyShimwa — APPROVED** [review]
Good Job 👏👏

**Review by jkarenzi — APPROVED** [review]

**Review by 13XAVI — APPROVED** [review]

File: index.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<!doctype html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <link rel="icon" type="image/svg+xml" href="/vite.svg" />
```
Comment (dushimeemma) [inline index.html]: design a favicon for our project

File: index.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Vite + React + TS</title>
```
Comment (dushimeemma) [inline index.html]: you can change the title as well.

File: public/vite.svg
Comment (dushimeemma) [inline public/vite.svg:1]: is this necessary?

File: src/assets/react.svg
Comment (dushimeemma) [inline src/assets/react.svg:1]: is this necessary?

File: index.html (hunk 1 above)
Comment (wayneleon1) [reply index.html]: icon-512

Comment (dushimeemma) [reply index.html]: Looks great @wayneleon1 💪

Comment (dushimeemma) [reply index.html]: is it possible to make it transparent?

Comment (wayneleon1) [reply index.html]: > is it possible to make it transparent?
yh , let me try

Comment (wayneleon1) [reply index.html]: icon-512-removebg-preview

Comment (wayneleon1) [reply index.html]: like that one

File: index.html (hunk 2 above)
Comment (wayneleon1) [reply index.html]: done

**Review by dushimeemma — APPROVED** [review]
Good job @wayneleon1 🥇

**Pull request merged**